pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -64,8 +81,26 @@ type stateData struct {
 	// might come from the informer cache. The instances get replaced when
 	// the plugin itself successfully does an Update.
 	//
-	// Empty if the Pod has no claims.
+	// In addition, the last claim in the slice may be the special claim for
+	// extended resource backed by DRA, which is created, updated, deleted by
+	// this plugin.
+	//
+	// Empty if the Pod has no claims and no special claim for extended
+	// resource backed by DRA.
 	claims []*resourceapi.ResourceClaim
+	// May have extended resource backed by DRA.
+	framework.Resource
+	// The mapping of extended resource to device class name
+	extendedResources map[v1.ResourceName]string
+	// UID of the special claim for extended resource backed by DRA.
+	// It is either the temporary UID for the in-memory claim, or
+	// it is the UID of the special claim written to API server in prior
+	// scheduling cycle, and found at PreFilter phase in this cycle.
+	extendedResourceClaimTplUID types.UID
+	// UID of the special claim for extended resource backed by DRA
+	// after it is written to API server at PreBind phase in this scheduling
+	// cycle.
+	extendedResourceClaimAPIUID types.UID
```

Comment:
I don't understand why you need to keep track of two UIDs for the special ResourceClaim.

This seems to be used to track whether the ResourceClaim actually exists. Can't you use the claim name and a bool for that instead? That would be much easier to understand. I'll stop reviewing the plugin code now because I hope that you can simplify it.

Reply:
This seems very indirect and hard to follow. My suggestion was to distinguish between "in memory" and "in API server" with a boolean.

I also don't get why you need to synthesize a UID when it's in memory.

Reply:
the code at various stages takes a claim, and it needs to inspect the claim to know whether the claim is the in-memory one, or one already persisted to API server, and based on this inspection result, the code takes different paths.

I assume the in-memory one should look the same as the regular claim (a claim that has already persisted to API server), so it is not special, or different in any way. Then all code that works with the in-memory claim should work well with the persisted claim. Are you suggesting to create the in-memory claim without a UID? That makes the in-memory claim special, I'm not sure if that's a good idea.